thread.c

```diff
@@ -609,6 +609,11 @@ thread_start_func_2(rb_thread_t *th, VALUE *stack_start, VALUE *register_stack_s
 		     th->abort_on_exception || RTEST(ruby_debug)) {
 		/* exit on main_thread */
 	    }
+	    else if (th->report_on_exception) {
+		void rb_threadptr_error_print(rb_thread_t *th, VALUE errinfo);
+		rb_threadptr_error_print(th, errinfo);
```

Comment:
This should also print the thread information (name, starting location) and say clearly it's an exception from a Thread, like in the feature:
```
Thread for block #<Proc:0x000000010d008a80@thread_error.rb:17> terminated with exception:
...
```